### PR TITLE
KeepCRV and fixes

### PR DESCRIPTION
**Pull request by matkam**
Notable changes:
- Default to Sushiswap instead of Uniswap
- KeepCRV logic
- Gas efficient migration
- Various small bug fixes

### CODE REVIEW

File: contracts/StrategyCurveEcrv.sol [hunk 1]
```diff
@@ -16,23 +16,29 @@ contract StrategyCurveEcrv is BaseStrategy {
     using SafeMath for uint256;
 
     address public constant gauge = address(0x3C0FFFF15EA30C35d7A85B85c0782D6c94e1d238);
+    address public constant voter = address(0xF147b8125d2ef93FB6965Db97D6746952a133934); // Yearn's veCRV voter
 
     address private uniswapRouter = 0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D;
     address private sushiswapRouter = 0xd9e1cE17f2641f24aE83637ab66a2cca9C378B9F;
 
-    address public crvRouter = 0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D;
+    address public crvRouter = 0xd9e1cE17f2641f24aE83637ab66a2cca9C378B9F; // default SushiSwap
```
Comment (poolpitako) [inline contracts/StrategyCurveEcrv.sol]: `address(0xd9e1cE17f2641f24aE83637ab66a2cca9C378B9F);`

File: contracts/StrategyCurveEcrv.sol [hunk 2]
```diff
@@ -56,34 +62,32 @@ contract StrategyCurveEcrv is BaseStrategy {
...
-
         uint256 gaugeTokens = proxy.balanceOf(gauge);
         if (gaugeTokens > 0) {
             proxy.harvest(gauge);
 
             uint256 crvBalance = crv.balanceOf(address(this));
-            if (crvBalance > 0) {
+            if (crvBalance > minToSwap) {
+                uint256 keepCrv = crvBalance.mul(keepCRV).div(FEE_DENOMINATOR);
+                IERC20(crv).safeTransfer(voter, keepCrv);
```
Comment (poolpitako) [inline contracts/StrategyCurveEcrv.sol]: shouldn't we have a setter for where the keepCRV is going?

**Review by poolpitako — APPROVED** [review]

File: contracts/StrategyCurveEcrv.sol [hunk 3]
```diff
@@ -104,38 +108,45 @@ contract StrategyCurveEcrv is BaseStrategy {
...
 
         return protected;
     }
 
     //
-    // helper functions
+    // setters
     //
-    function setCRVRouter(bool isUniswap, address[] calldata _wethPath) public onlyGovernance {
+    function setCRVRouter(bool isUniswap, address[] calldata _wethPath) external onlyAuthorized {
```
Comment (storming0x) [inline contracts/StrategyCurveEcrv.sol]: Remove the calldata _wethPath argument, just leaving the feature of switching dexes, path is the same on both uni and sushi. And contract can be migrated as well

File: tests/test_ecrv_live.py [hunk 4]
```diff
@@ -66,23 +82,37 @@ def test_migrate_live(
...
+    vault_ecrv_live.updateStrategyDebtRatio(strategy_ecrv_live, 0, {"from": gov_live})
+    strategy_ecrv_live.harvest({"from": dev})
+    vault_ecrv_live.migrateStrategy(strategy_ecrv_live, strategy_ecrv2, {"from": gov_live})
+
+    gauge = "0x3C0FFFF15EA30C35d7A85B85c0782D6c94e1d238"
+    voter_proxy.approveStrategy(gauge, strategy_ecrv2, {"from": gov_live})
+
     genericStateOfStrat(strategy_ecrv_live, token_ecrv, vault_ecrv_live)
     genericStateOfStrat(strategy_ecrv2, token_ecrv, vault_ecrv_live)
     genericStateOfVault(vault_ecrv_live, token_ecrv)
```
Comment (storming0x) [inline tests/test_ecrv_live.py:80]: I would suggest adding balance checks before and after migration, seems like genericState prints the info,
but the values should be asserted against to be sure.

**Pull request merged**